force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,36 @@
+trigger OpportunityTrigger on Opportunity (before update, before delete) {
+    if (Trigger.isBefore && Trigger.isUpdate) {
+        Set<Id> accountIds = new Set<Id>();
+        for (Opportunity opp : Trigger.new) {
+            if (opp.Amount < 5000) {
+                opp.addError('Opportunity amount must be greater than 5000');
+            }
+            accountIds.add(opp.AccountId);
+        }
+        Set<Contact> contactsToUpdate = new Set <Contact>([SELECT Id, AccountId FROM Contact WHERE Title = 'CEO' AND AccountId IN :accountIds]);
+        for(Contact con : contactsToUpdate){
+            for(Opportunity opp : Trigger.new){
+                if(opp.Primary_Contact__c == null){
+                    opp.Primary_Contact__c = con.Id;
```

Comment:
Well done setting the primary contact. I would again advise moving the actual business logic to a handler or service class